files.sh

```diff
@@ -2,7 +2,7 @@
 
 fn nash_complete_paths(parts, line, pos) {
 	partsz   <= len($parts)
-	last     <= -expr $partsz - 1
+	last     <= -echo $partsz - 1 | bc
```

Comment:
Looking at it now I really think this could be a bug in nash. Take a look at the test I did:
```sh
[«~»]λ> last <= -expr 1 - 1
[«~»]λ> echo $last
ERROR: <interactive>:1:5: Variable $last not set on shell parent scope
[«~»]λ> last <= -expr 2 - 1
[«~»]λ> echo $last
1
[«~»]λ> last <= -expr 1 - 1
[«~»]λ> echo $last
1
[«~»]λ> 
```

But if we just execute `-expr 1 - 1` the result is `0`. 

So my first guess would be to say nash is not assigning the value `0` to variables when there is an error.